### ISSUE TITLE
Newlines in logs

### ISSUE THREAD

**Issue by matejzero** [skx/puppet-summary#40]
Hello,

first, thank you for this great tool. I'm just testing it out now as a replacement for our puppet-dashboard and it's looking great:)

One thing I noticed is that it doesn't take new lines into account when showing logs. This would be useful when looking at diffs of files.

Current output:
```
--- /etc/my.cnf.d/server.cnf	2017-12-29 13:29:49.617471149 +0100 +++ /tmp/puppet-file20180305-43916-149xpky	2018-03-05 09:16:57.740678088 +0100 @@ -49,6 +49,7 @@ read-only = ON relay_log = /var/lib/mysql/data/relay-bin relay_log_index = /var/lib/mysql/data/relay-bin.index +replicate-wild-ignore-table = otrs.% report_host = foo.bar.org server_id = 13 slow_query_log = 1
```

I think it would be really nice to have some structure, so it would be easier to see changes.

**Comment by skx** [comment]
Yes I've noticed the same thing.  It's a frustrating issue because the obvious solution is just to wrap the output in a `<pre>` + `</pre>` block.  But that looks bad for some cases too.

I've never quite sat down and gone through all the output but I think that it is probably worth defaulting to the wrapping, and then selectively disabling it.

**Comment by skx** [comment]
If you can build from source - and regenerate the templates - then this change should be all that is required:

      diff --git a/data/report.template b/data/report.template
      index 83e0c2f..6626274 100644
      --- a/data/report.template
      +++ b/data/report.template
      @@ -63,7 +63,7 @@
                 <div class="col-sm-1 col-md-1">
                 </div>
                 <div class="col-sm-11 col-md-11">
      -            {{range .LogMessages}}
      +            <pre>{{range .LogMessages}}</pre>
                   <p>{{.}}</p>
                   {{else}}
                   <p>Nothing reported.</p>

I'll commit that as a new release this evening.

**Comment by matejzero** [comment]
Sure, I'll give it a go and report:)

**Comment by matejzero** [comment]
This looks way better. But I needed to set `<pre>` one line lower:

```
  diff --git a/data/report.template b/data/report.template
  index 83e0c2f..6626274 100644
  --- a/data/report.template
  +++ b/data/report.template
  @@ -63,7 +63,7 @@
             <div class="col-sm-1 col-md-1">
             </div>
             <div class="col-sm-11 col-md-11">
              {{range .LogMessages}}
  -            <p>{{.}}</p>
  +            <pre><p>{{.}}</p></pre>
               {{else}}
               <p>Nothing reported.</p>
```

We could do some CSS modification as well to lower the line-heigh inside `<pre>` to get rid of extra blank space under the text:P.

**Comment by skx** [comment]
D'oh.  That's what happens when I make a quick diff without compiling/testing!

CSS change is probably reasonable too :)

**Comment by matejzero** [comment]
Will take a look at CSS and try to create something nice:P

**Comment by matejzero** [comment]
I see you are loading CSS from CDN. What would be the best way to do modifications? Inline CSS or extra .css file?

As far as changes go, it looks OK to me if I change:
```
p:
  margin: 0; (default 0 0 10px)
pre:
  padding: 5px 9px; (default 9,5px)
```

**Comment by skx** [comment]
I'd probably place it in-line to avoid having to write a handler to serve "`/extra.css`".

(If the change were more general I'd probably make the other decision, but since this is only a change useful for `pre` tags, which will only show up in the report-view I think inline is fine.)

**Comment by matejzero** [comment]
Nice.

If you don't mind, I can create a PR.

**Comment by skx** [comment]
Please do feel free.  I'll not be home for a good few hours and if there is one waiting for me I can bundle it up with a new release quickly enough.

**Comment by matejzero** [comment]
PR created.

I've testing it and it looks OK on my side. Try it yourself too and merge if you think it's good:)

**Comment by skx** [comment]
The change looks good - but it doesn't show up as a PR against _this_ repository...

**Comment by matejzero** [comment]
Sorry, I just edited the file via web and didn't look where PR landed. Fixed that.

**Comment by skx** [comment]
Closed in #41 - much appreciated.  I'll sort out a new release shortly.
